Client/Telemetry/TelemetryWrapper.swift

```diff
@@ -377,65 +377,59 @@ extension TelemetryWrapper {
+            GleanMetrics.Bookmarks.viewList[from].add()
+        case (.action, .add, .bookmark, let from, _):
+            GleanMetrics.Bookmarks.add[from].add()
+        case (.action, .delete, .bookmark, let from, _):
+            GleanMetrics.Bookmarks.delete[from].add()
+        case (.action, .open, .bookmark, let from, _):
+            GleanMetrics.Bookmarks.open[from].add()
         // Reader Mode
         case (.action, .tap, .readerModeOpenButton, _, _):
             GleanMetrics.ReaderMode.open.add()
         case (.action, .tap, .readerModeCloseButton, _, _):
             GleanMetrics.ReaderMode.close.add()
         // Reading List
-        case (.action, .add, .readingListItem, let value, _):
-            GleanMetrics.ReadingList.add[value].add()
-        case (.action, .delete, .readingListItem, let value, _):
-            GleanMetrics.ReadingList.delete[value].add()
+        case (.action, .add, .readingListItem, let from, _):
+            GleanMetrics.ReadingList.add[from].add()
+        case (.action, .delete, .readingListItem, let from, _):
+            GleanMetrics.ReadingList.delete[from].add()
         case (.action, .open, .readingListItem, _, _):
             GleanMetrics.ReadingList.open.add()
         case (.action, .tap, .readingListItem, EventValue.markAsRead.rawValue, _):
             GleanMetrics.ReadingList.markRead.add()
         case (.action, .tap, .readingListItem, EventValue.markAsUnread.rawValue, _):
             GleanMetrics.ReadingList.markUnread.add()
         // Preferences
-        case (.action, .change, .setting, let preference, let extras):
-            if let to = extras?["go"] as? String {
+        case (.action, .change, .setting, _, let extras):
+            if let preference = extras?["pref"] as? String, let to = (extras?["go"] ?? "undefined") as? String {
                 GleanMetrics.Preferences.changed.record(
                 extra: [GleanMetrics.Preferences.ChangedKeys.preference: preference,
                         GleanMetrics.Preferences.ChangedKeys.changedTo: to])
             } else {
-                GleanMetrics.Preferences.changed.record(
-                    extra: [GleanMetrics.Preferences.ChangedKeys.preference: preference,
-                            GleanMetrics.Preferences.ChangedKeys.changedTo: "undefined"])
+                let msg = "Uninstrumented pref metric: \(category), \(method), \(object), \(value), \(String(describing: extras))"
+                Sentry.shared.send(message: msg, severity: .debug)
             }
         // QR Codes
         case (.action, .scan, .qrCodeText, _, _),
              (.action, .scan, .qrCodeURL, _, _):
             GleanMetrics.QrCode.scanned.add()
         // Tabs
-        case (.action, .add, .tab, let value, _):
-            GleanMetrics.Tabs.open[value].add()
-        case (.action, .close, .tab, let value, _):
-            GleanMetrics.Tabs.close[value].add()
+        case (.action, .add, .tab, let privateOrNormal, _):
```

Comment:
the only possible values are 'private' and 'normal'. I thought maybe I should not convert the EventValue to a string on line 387 and then we could make it clearer what enum values we are handling in each case statement. But that would mean a bunch more changes to this switch statement.